gnucash/gtkbuilder/window-autoclear.glade

```diff
@@ -71,16 +68,59 @@
               <object class="GtkLabel" id="label1">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
-                <property name="label" translatable="yes">&lt;b&gt;Auto-Clear Information&lt;/b&gt;
-Use this dialog if you want GnuCash to automatically find which transactions are cleared, given an ending balance. For example, said ending balance can be the current balance given by your online bank.</property>
-                <property name="use_markup">True</property>
+                <property name="label" translatable="yes">About Auto-Clear</property>
+                <style>
+                  <class name="gnc-class-highlight"/>
+                </style>
+              </object>
+              <packing>
+                <property name="expand">False</property>
+                <property name="fill">False</property>
+                <property name="position">0</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkLabel" id="label2">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="label" translatable="yes">Use this dialog if you want GnuCash to automatically find which transactions are cleared, given an ending balance. For example, said ending balance can be the current balance given by your bank online.</property>
                 <property name="wrap">True</property>
                 <property name="max_width_chars">80</property>
               </object>
               <packing>
                 <property name="expand">False</property>
                 <property name="fill">False</property>
-                <property name="position">0</property>
+                <property name="position">1</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkLabel" id="label3">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="has_tooltip">True</property>
+                <property name="label" translatable="yes">Caution!</property>
+                <style>
+                  <class name="gnc-class-emphasize-label"/>
+                </style>
+              </object>
+              <packing>
+                <property name="expand">False</property>
+                <property name="fill">False</property>
+                <property name="position">2</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkLabel" id="label4">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="label" translatable="yes">The complexity of this algorithm is quite high. Use this only on a dozen of uncleared splits.</property>
```

Comment:
From https://github.com/Gnucash/gnucash/pull/805#discussion_r515151222

The caution does not need to be too specific; it could state "Caution! The complexity of this algorithm is quite high; it may abort if the search size becomes excessive."

Reply:
> it may abort if the search size becomes excessive.

Can you include a magnitude? Worst case it will abort for …

Reply:
As described worst case scenario is 1mil amounts reached after 20 splits.

Reply:
How about "Caution! The complexity of this algorithm is quite high; it may abort if the search size becomes excessive. Worst case the limit can be reached with 20 splits."?

Reply:
Sure, that's fine

Reply:
ISTM "The complexity of this algorithm is high" will be meaningless to the users who aren't coders; it might even be meaningless to some of the coders.

Something along the lines of "This feature can take a very long time if there are 20 or more uncleared splits." would be easier for the general user to understand.

Reply:
> "This feature can take a very long time if there are 20 or more uncleared splits."

I think this is mixing 2 different things

1. It can take a very long time,

1. It can abort worst case for 20 or more uncleared splits.

Reply:
> "This feature can take a very long time if there are 20 or more uncleared splits."

Aren't you now mixing 2 different things

1. It can take a very long time,

1. It can worst case abort if there are 20 or more uncleared splits.

Reply:
Strange, my answer had somehow disappeared and now it is there again.

Reply:
Not really, though it perhaps mis-states that a largish number of uncleared splits somewhere around 20 will cause it to abort after taking a really long time. The actual abort occurs when the number of combinations reaches 1 million.

In any case, you're missing the point: Regardless of the actual message, the message should be understandable by users. The current wording makes sense only to programmers.

Reply:
Actually I had a failsafe; it won't ever take too long. The most accurate statement is: "this feature is not enabled if the number of uncleared splits is more than approximately 20. If this is the case please reconcile manually."

Reply:
> this feature is not enabled

Really, or "not recommend"?
Is there a better word for "feature"?

Reply:
Try an account with 20+ uncleared splits, amounts 1, 2, 4, 8 etc up to around 1048576. It should fail to generate a split list to clear (without a long delay) unless @cristiklein or anyone can find a better algorithm.

Reply:
How embarrassing. On testing with real data, with splits to 2^18 = $131072 onwards there is definitely slowdown. So, my #813 is not that clever. Either I'll rework it to cache `sack` instead of `toclear_list`, or revert it altogether.

Reply:
> "this feature is not enabled if the number of uncleared splits is more than approximately 20. If this is the case please reconcile manually."

How about `This tool is not recommend if the number of uncleared splits is more than approximately 20. In that case please reconcile at least some of them manually.`

Reply:
"clear" rather than "reconcile". "not recommend[ed]" doesn't really explain anything: "might be slow" and unless @christopherlam reverts his hack "won't work" are clearer.